MyCryptoHQ/MyCrypto
# Node Refactor (Pt. 2 - Web Node Selector Rework)

Closes the web portion of #1353 & #1455.

### Description

This rebuilds the web network selector to be a totally custom component that emphasizes a network-first view of nodes. When opening the dropdown now, you'll be presented with each network as an option, instead of all of the nodes. Selecting a network will choose the auto config for that network. You can expand each network to be a list of nodes, and select individuals from there.

Includes some minor refactors of types, as well as some change in behavior for custom nodes.

### Changes

* Change the node selector from a ColorDropdown to a custom component called NetworkSelector (Gif below)
    * Presents networks as options, individual nodes hidden in collapsibles
    * Changes auto network displays to show `NETWORK (X Nodes)` instead of `NETWORK (AUTO)`
* Removed the ColorDropdown component
* Added changeNetworkIntent action that selects the auto node for a given network, or the first node if it has no auto (e.g. custom network)
    * My vision for the future is that this dropdown is _only_ network selection, and that you specify on an options page which nodes a given network should use. So `changeNode*` functions will disappear, and we'll only change networks, which will be configured independently.
    * In a future refactor PR, I'd like to make `node.name` be display-only, and instead us `node.id` for logic stuff. Right now we use `node.name` to identify static nodes, and `node.chainId` to identify custom nodes. This is a huge pain in the ass.
* Added an `id` attribute to `StaticNodeConfig`s that matches their key on the staticNodes reducer
* Fix `removeCustomNode` from resetting you to ETH network all the time

### Steps to Test

1. Open up the new network selector
2. Click on a network, confirm you are on the auto node of that network
3. Expand the dropdown of a network and select an individual node, confirm you are only connected to that node
4. Add a custom node, confirm it shows up in the correct network
5. Add a custom node with a custom network, confirm the network shows up with that node
6. Remove the custom node, confirm it no longer shows up in the menu


## Screenshots

![network-selector](https://user-images.githubusercontent.com/649992/39020808-1cb32aa4-43fc-11e8-9832-1e942c9d05c1.gif)
